compiler/rustc_middle/src/query/erase.rs

```diff
@@ -0,0 +1,234 @@
+use crate::mir;
+use crate::traits;
+use crate::ty;
+use rustc_ast::expand::allocator::AllocatorKind;
+use rustc_hir::MaybeOwner;
+pub use rustc_middle::traits::query::type_op;
+use std::intrinsics::type_name;
+use std::{
+    fmt,
+    mem::{size_of, transmute_copy, MaybeUninit},
+};
+
+#[derive(Copy, Clone)]
+pub struct Erased<T: Copy> {
+    data: MaybeUninit<T>,
+    #[cfg(debug_assertions)]
+    type_id: &'static str,
```

Comment:
Why is this a `&str` instead of a real `TypeId`?

Reply:
Because that requires `T: 'static`

Reply:
Oh, I thought `type_name` did too.